File: src/main/org/nlogo/api/GraphicsInterface.java
package org.nlogo.api;

// all renderer classes deal only with GraphicsInterfaces rather
// than Graphics2D objects so we can test rendering

public interface GraphicsInterface {
  void antiAliasing(boolean on);

  void draw(java.awt.Shape shape);

  void drawImage(java.awt.image.BufferedImage image);

  void drawImage(java.awt.Image image, int x, int y, int width, int height);

  void drawLine(double x1, double y1, double x2, double y2);

  void drawLabel(String s, double x, double y, double patchSize);

  void fill(java.awt.Shape shape);

  void fillRect(int x, int y, int width, int height);

  void pop();

  void push();

  void rotate(double theta);

  void rotate(double theta, double x, double y);

  void rotate(double theta, double x, double y, double offset);

  void scale(double x, double y);

  void scale(double x, double y, double shapeWidth);

  void setColor(java.awt.Color c);

  void setComposite(java.awt.Composite comp);

  void setStroke(double width);

  void setStroke(float width, float[] dashes);

  void setStrokeFromLineThickness(double lineThickness, double scale, double cellSize, double shapeWidth);

  void translate(double x, double y);

  void setInterpolation();

  void setStrokeControl();

  void drawPolygon(int[] xcors, int ycors[], int length);

  void fillPolygon(int[] xcors, int ycors[], int length);

  void drawPolyline(int[] xcors, int ycors[], int length);

  void dispose();

  String location(double x, double y);

  void fillCircle(double x, double y, double xDiameter, double yDiameter, double scale, double angle);

  void drawCircle(double x, double y, double xDiameter, double yDiameter, double scale, double angle);

  void fillRect(double x, double y, double width, double height, double scale, double angle);

  void drawRect(double x, double y, double width, double height, double scale, double angle);

  java.awt.FontMetrics getFontMetrics();
}

File: src/main/org/nlogo/api/GraphicsInterface.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

// so we can mock Graphics2D for testing

trait GraphicsInterface {
  def antiAliasing(on: Boolean)
  def draw(shape: java.awt.Shape)
  def drawImage(image: java.awt.image.BufferedImage)
  def drawImage(image: java.awt.Image, x: Int, y: Int, width: Int, height: Int)
  def drawLine(x1: Double, y1: Double, x2: Double, y2: Double)
  def drawLabel(s: String, x: Double, y: Double, patchSize: Double)
  def fill(shape: java.awt.Shape)
  def fillRect(x: Int, y: Int, width: Int, height: Int)
  def pop()
  def push()
  def rotate(theta: Double)
  def rotate(theta: Double, x: Double, y: Double)
  def rotate(theta: Double, x: Double, y: Double, offset: Double)
  def scale(x: Double, y: Double)
  def scale(x: Double, y: Double, shapeWidth: Double)
  def setColor(c: java.awt.Color)
  def setComposite(comp: java.awt.Composite)
  def setStroke(width: Double)
  def setStroke(width: Float, dashes: Array[Float])
  def setStrokeFromLineThickness(lineThickness: Double, scale: Double, cellSize: Double, shapeWidth: Double)
  def translate(x: Double, y: Double)
  def setInterpolation()
  def setStrokeControl()
  def drawPolygon(xcors: Array[Int], ycors: Array[Int], length: Int)
  def fillPolygon(xcors: Array[Int], ycors: Array[Int], length: Int)
  def drawPolyline(xcors: Array[Int], ycors: Array[Int], length: Int)
  def dispose()
  def location(x: Double, y: Double): String
  def fillCircle(x: Double, y: Double, xDiameter: Double, yDiameter: Double, scale: Double, angle: Double)
  def drawCircle(x: Double, y: Double, xDiamter: Double, yDiameter: Double, scale: Double, angle: Double)
  def fillRect(x: Double, y: Double, width: Double, height: Double, scale: Double, angle: Double)
  def drawRect(x: Double, y: Double, width: Double, height: Double, scale: Double, angle: Double)
  def getFontMetrics: java.awt.FontMetrics
}

